Upgrades tc-events to new libraries and node 4

ckousik opened the pull request:
```
modified:   README.md
modified:   bin/server.js
new file:   config.yml
deleted:    config/defaults.js
deleted:    config/localhost.js
deleted:    config/production.js
modified:   package.json
new file:   test/events.js
modified:   test/runtests.sh
```

ckousik committed:
Upgrades tc-events to new libraries and node 4

	modified:   README.md
	modified:   bin/server.js
	new file:   config.yml
	deleted:    config/defaults.js
	deleted:    config/localhost.js
	deleted:    config/production.js
	modified:   package.json
	new file:   test/events.js
	modified:   test/runtests.sh

ckousik committed:
Upgrades tc-events to new libraries and node 4

	modified:   README.md
	modified:   bin/server.js
	new file:   config.yml
	deleted:    config/defaults.js
	deleted:    config/localhost.js
	deleted:    config/production.js
	modified:   package.json
	new file:   test/events.js
	modified:   test/runtests.sh

	modified:   .travis.yml
	modified:   config.yml
	modified:   test/events.js

	modified:   .travis.yml
	modified:   config.yml
	modified:   package.json

	modified:   test/events.js

Fixes the tests
	modified:   test/events.js

Fixes to server, test, config.yml
Ran shrinkwrap
Edited travis.yml to install build-essential
	modified:   .gitignore
	modified:   .travis.yml
	modified:   bin/server.js
	modified:   config.yml
	new file:   npm-shrinkwrap.json
	renamed:    test/events.js -> test/event_test.js
	modified:   test/runtests.sh

Test out travis.yml fix
	modified:   .travis.yml

Removed sudo from .travis.yml
	modified:   .travis.yml

Yet another .travis.yml fix
	modified:   .travis.yml

Run node 4...travis.yml fix
	modified:   .travis.yml

Update config.yml
	modified:   config.yml

Fixed env vars problem
	modified:   bin/server.js
	modified:   config.yml
	modified:   test/event_test.js

Final changes to config.yml and server.js

Upgrade all dependencies and switch to node 4
	modified:   .gitignore
	modified:   .travis.yml
	modified:   README.md
	modified:   bin/server.js
	new file:   config.yml
	deleted:    config/defaults.js
	deleted:    config/localhost.js
	deleted:    config/production.js
	new file:   npm-shrinkwrap.json
	modified:   package.json
	modified:   test/event_test.js
	modified:   test/runtests.sh

Add user-config-example.yml

Upgrade dependencies and switch to node v4
	modified:   .gitignore
	modified:   .travis.yml
	modified:   README.md
	modified:   bin/server.js
	new file:   config.yml
	deleted:    config/defaults.js
	deleted:    config/localhost.js
	deleted:    config/production.js
	new file:   npm-shrinkwrap.json
	modified:   package.json
	modified:   test/event_test.js
	modified:   test/runtests.sh
	new file:   user-config-example.yml

ckousik committed:
modified:   .travis.yml
	modified:   config.yml
	modified:   test/events.js

ckousik committed:
modified:   .travis.yml
	modified:   config.yml
	modified:   package.json

ckousik committed:
modified:   test/events.js

Fixes the tests
	modified:   test/events.js

jonasfj commented on config.yml:
```diff
@@ -0,0 +1,35 @@
+  taskcluster:
+    credentials:
+      accessToken:    !env TASKCLUSTER_CREDENTIALS_ACCESSTOKEN
+      clientId:       !env TASKCLUSTER_CREDENTIALS_CLIENTID
```
These should be: TASKCLUSTER_CLIENT_ID and TASKCLUSTER_ACCESS_TOKEN like everywhere else..

jonasfj replied:
@ckousik  shouldn't have changed this for tests to pass?

ckousik replied:
I added a new commit...I'll squash them later

jonasfj commented on bin/server.js:
```diff
@@ -1,62 +1,52 @@
-      'pulse_username',
-      'pulse_password'
+    files: [
+      path.join(__dirname,'../config.yml')
```
is this really necessary?

jonasfj replied:
look at the defaults:
https://github.com/taskcluster/typed-env-config/blob/master/src/config.js#L48-L55

Just make sure current working folder is the top-level directory...

note that: `process.env.NODE_ENV` is "production" in heroku. so we only really need to pass it in when we're in testing...

What is the use for overwriting env... I think we can avoid that..

jonasfj commented on bin/server.js:
```diff
@@ -1,62 +1,52 @@
-    process:    'server'
-  });
+  var influx = null;
+  if(cfg.influx.connectionString){
```
let's just remove all influx usage...

ckousik committed:
Fixes to server, test, config.yml
Ran shrinkwrap
Edited travis.yml to install build-essential
	modified:   .gitignore
	modified:   .travis.yml
	modified:   bin/server.js
	modified:   config.yml
	new file:   npm-shrinkwrap.json
	renamed:    test/events.js -> test/event_test.js
	modified:   test/runtests.sh

ckousik committed:
Test out travis.yml fix
	modified:   .travis.yml

ckousik committed:
Removed sudo from .travis.yml
	modified:   .travis.yml

ckousik committed:
Yet another .travis.yml fix
	modified:   .travis.yml

ckousik committed:
Run node 4...travis.yml fix
	modified:   .travis.yml

jonasfj commented on bin/server.js:
```diff
@@ -1,62 +1,41 @@
-      'pulse_password'
+    files: [
+      'config.yml',
+      'user-config.yml'
```
I think we can just leave this empty? as this is this default value, right?

ckousik committed:
Update config.yml
	modified:   config.yml

ckousik committed:
Fixed env vars problem
	modified:   bin/server.js
	modified:   config.yml
	modified:   test/event_test.js

jonasfj commented:
# `r+`

imbstack commented on README.md:
````diff
@@ -45,6 +45,12 @@ listener.on('close', function() {
 ```
+Testing
+-------
+1. Add pulse credentials to test profile in config.yml
````
These should be in `user-config.yml` instead of `config.yml`. I see you added it to the `.gitignore` so I think you meant to say it here as well.

Whenever there's a `user-config.yml` to be used, I like to have a `user-config-example.yml` checked into the project.

imbstack commented on bin/server.js:
```diff
@@ -1,62 +1,35 @@
-    trustProxy:     cfg.get('server:trustProxy')
+  var app = new base.app({
+    port:               Number(process.env.PORT || cfg.port),
+    forceSSL:           cfg.forceSSL ||false,
```
I don't think we need to have `|| false` here. It should do the right thing without that. If either the `base.config` or `base.app` don't handle this correctly, I'd consider it a bug.

imbstack commented on config.yml:
```diff
@@ -0,0 +1,35 @@
+  port:               12345
+  forceSSL:           false
+  pulse:
+    username:         !env PULSE_USERNAME
```
These aren't necessary. It will just use whatever was in defaults if left unspecified. Move the corresponding section in the production set into default and you'll be all set.

imbstack commented on test/event_test.js:
```diff
@@ -1,221 +1,108 @@
-  var taskcluster = require('taskcluster-client');
-  var debug       = require('debug')('test:event');
-  var base        = require('taskcluster-base');
+'use strict'
```
Since we're transpiling with babel, we don't really want to specify `use strict` here, right? The transpiler will handle it for us.

ckousik replied:
I was having some problems  when not using 'use strict'

imbstack replied:
Weird. What were the problems like?

ckousik replied:
It wouldn't let me use `let`

imbstack replied:
Ah, we're not even using babel in this project at all. This is fine for now, but we should add babel soon.

imbstack commented:
Looking good so far!

imbstack commented on README.md:
```diff
@@ -45,6 +45,12 @@ listener.on('close', function() {
+Testing
+-------
+1. Add pulse credentials to test profile in config.yml
+2. Add taskcluster credentials to test profile in config.yml. Ensure you have the necessary scopes.
```
It would be nice if this specified either the necessary scopes or a role that can be assumed.

imbstack commented:
Looks good to me. r+, fix as many nits as you want and then we can ship this.

ckousik committed:
Merge branch 'update-node' of https://github.com/ckousik/taskcluster-events into update-node

Conflicts:
	README.md
	bin/server.js
	config.yml

ckousik committed:
config.yml
	modified:   config.yml

ckousik committed:
Added bind test and changed config.yml
	modified:   config.yml
	modified:   test/event_test.js
	modified:   test/runtests.sh
